### PR TITLE
Rewrites the BccToggle component as a CSS library

### PR DESCRIPTION
**Pull request by mvelzel**
This pull request also adds the `withIcon` property which allows the option of enabling or disabling the icons displayed in the toggle. This is `true` by default as to not break existing applications using this toggle.
It also adds the `label` property according to the design to allow the option of adding a label to the toggle. This is empty by default as to not break existing applications.

Other than those two new additions, all functionality is the same as the previous toggle, while having removed Headless UI as a dependency.

### CODE REVIEW

**Review by laurensgroeneveld — APPROVED** [review]
LGTM

**Comment by u12206050** [comment]
What is the reason for removing the use of Headless ui? That was what was adding a lot of accessibility features which I don't see have been reimplemented in the latest code. @laurensgroeneveld

**Comment by laurensgroeneveld** [comment]
> What is the reason for removing the use of Headless ui? That was what was adding a lot of accessibility features which I don't see have been reimplemented in the latest code. @laurensgroeneveld

Accessibility is a fair concern and something we'd need to check, although I would expect that by using a native checkbox a lot of that is covered.

The reason behind this change was to make this a CSS component so it can be used in the CSS library. We decided that in this case HeadlessUI didn't provide features that we needed and made it more or less impossible to add reusable CSS classes to this component, and also because it can be implemented in CSS. In the case of something like a dropdown where we do need custom Javascript anyways we'll still use HeadlessUI.

**Comment by u12206050** [comment]
I think maybe that there is a misunderstanding of what headlessui provides.
It only provides functionality and no design.
The css classes add design, and might to a certain degree add
functionality, however most functionality will be either have to be added
by the user using the classes or it should already be added in the vue
components.

So the fact that we want css classes doesn't remove the fact we need logic
and accessibility features, which headless ui offers.

**Comment by mvelzel** [comment]
But in this case it is completely unnecessary. This implementation stays as close as possible to pure HTML, and using an actual `<input type="checkbox" />` is in every single way more accessible than using a `<button>` with some added CSS, which is what Headless UI does.
Aside from that there is also no issue when it comes to logic issues because Vue obviously has great support for pure HTML inputs and its logic.

**Comment by mvelzel** [comment]
Aside from those arguments, my personal argument is that I want to use the CSS library outside of Vue, but if Headless UI still becomes a requirement for using the CSS library then I would resort to not using it at all.
To support the use case of being able to use the CSS library without the need for additional JS libraries we need to rely on the functionality HTML provides with their `<input type="checkbox" />` instead of the functionality that Headless UI reimplements using JS.

Basically if it was not implemented like this I would have no use for the CSS library in the case of the CSS toggle in my use-case.

**Comment by mvelzel** [comment]
Of course with certain components like a dropdown you cannot get around the Headless UI requirement. But in cases where we can such as here, it increases the adoptability of the CSS library.

**Comment by u12206050** [comment]
Ok, makes sense.
